src/main/java/com/mraof/minestuck/network/EditmodeDragPacket.java

```diff
@@ -0,0 +1,342 @@
+	
+	private static boolean editModePlaceCheck(Level level, Player player, InteractionHand hand)
+	{
+		if(!level.isClientSide() && ServerEditHandler.getData(player) != null)
+		{
+			EditData data = ServerEditHandler.getData(player);
+			SburbConnection connection = data.getConnection();
+			
+			ItemStack stack = player.getMainHandItem();
+			
+			if(stack.isEmpty() || hand.equals(InteractionHand.OFF_HAND))
+				return false;
+			
+			DeployEntry entry = DeployList.getEntryForItem(stack, connection, level);
+			if(entry != null)
+			{
+				GristSet cost = entry.getCurrentCost(connection);
+				if(!GristHelper.canAfford(PlayerSavedData.getData(connection.getClientIdentifier(), level).getGristCache(), cost))
+					return false;
+				
+			}
+			else if(!(stack.getItem() instanceof BlockItem) || !GristHelper.canAfford(level, connection.getClientIdentifier(), GristCost.findCostForItem(stack, null, false, level)))
+				return false;
+		}
+		
+		return true;
+	}
+	
+	private static GristSet editModePlaceCost(Level level, Player player, InteractionHand hand, BlockPos pos)
+	{
+		EditData data = ServerEditHandler.getData(player);
+		SburbConnection connection = data.getConnection();
+		
+		ItemStack stack = player.getMainHandItem();
+		
+		if(stack.isEmpty() || hand.equals(InteractionHand.OFF_HAND))
+			return new GristSet();
+		
+		DeployEntry entry = DeployList.getEntryForItem(stack, connection, level);
+		if(entry != null)
+		{
+			GristSet cost = entry.getCurrentCost(connection);
+			if(level.getBlockState(pos).getMaterial().isReplaceable() && !GristHelper.canAfford(PlayerSavedData.getData(connection.getClientIdentifier(), level).getGristCache(), cost))
+				return cost;
+			
+		}
+		else if(level.getBlockState(pos).getMaterial().isReplaceable() && !GristHelper.canAfford(level, connection.getClientIdentifier(), GristCost.findCostForItem(stack, null, false, level)))
+			return GristCost.findCostForItem(stack, null, false, level);
+		
+		return new GristSet();
+	}
+	
+	private static boolean editModeDestroyCheck(Level level, Player player, BlockPos pos)
+	{
+		if(!level.isClientSide() && ServerEditHandler.getData(player) != null)
+		{
+			EditData data = ServerEditHandler.getData(player);
+			SburbConnection connection = data.getConnection();
+			
+			BlockState block = level.getBlockState(pos);
+			ItemStack stack = block.getCloneItemStack(null, level, pos, player);
+			DeployEntry entry = DeployList.getEntryForItem(stack, data.getConnection(), level, DeployList.EntryLists.ATHENEUM);
+			if(block.isAir())
+				return false;
+			else if(!MinestuckConfig.SERVER.gristRefund.get() && entry == null)
+			{
+				
+				GristSet cost = new GristSet(GristTypes.BUILD,1);
+				if(!GristHelper.canAfford(PlayerSavedData.getData(connection.getClientIdentifier(), level).getGristCache(), cost))
+					return false;
+				
+			}
+		}
+		return true;
+	}
+	
+	private static GristSet editModeDestroyCost(Level level, Player player, BlockPos pos)
```

Comment:
Seeing some overlap with `editModeDestroyCheck()`. Perhaps the code should be structured slightly differently?